Android/BacktraceCrashHandler.java

```diff
@@ -0,0 +1,41 @@
+package backtraceio.library.nativeCalls;
+
+import android.util.Log;
+
+import java.util.Map;
+
+public class BacktraceCrashHandler {
+    private static final String LOG_TAG = BacktraceCrashHandler.class.getSimpleName();
+    
+    public static final String BACKTRACE_CRASH_HANDLER = "BACKTRACE_UNITY_CRASH_HANDLER";
+
+
+    private static native boolean handleCrash(String[] args);
+
+    public static void main(String[] args) {
+        run(args, System.getenv());
+    }
+
+    public static boolean run(String[] args, Map<String, String> environmentVariables) {
+        if (environmentVariables == null) {
+            Log.e(LOG_TAG, "Cannot capture crash dump. Environment variables are undefined");
+            return false;
+        }
+
+        String crashHandlerLibrary = environmentVariables.get(BACKTRACE_CRASH_HANDLER);
+        if (crashHandlerLibrary == null) {
+            Log.e(LOG_TAG, String.format("Cannot capture crash dump. Cannot find %s environment variable", BACKTRACE_CRASH_HANDLER));
+            return false;
+        }
+        System.load(crashHandlerLibrary);
+
+        boolean result = handleCrash(args);
```

Comment:
does it make sense to try/catch this?

Reply:
this code shouldn't crash. in addition to that, we cannot do anything with an exception, unless you have an idea what to do. 
This
 code invokes crashpad, and our native interface provides a boolean response. If anything happens in the NDK layer, most likely it will crash the app, so try/catch won't handle it.

Also - since this is a class invoked by the external command, it would be better if we provide a full description of the exception with parameters. Wrapping it into a try/catch block with some nice messages won't help.